### DISCUSSION TITLE
io.micronaut.jaxrs.runtime.core.JaxRsRuntime Not getting loaded in IntelliJ.

### DISCUSSION THREAD

**Discussion by subash89** [micronaut-projects/micronaut-core#10843] [Q&A] [answered]
I am using IntelliJ IDEA 2024.1 with maven 3.9.5 and micronaut 4.4.2 micronaut-parent.

When running my micronaut jaxrs app in IntelliJ I am getting a 404 for my apis. When looked at TRACE logs when that happens I see following.

```
console: 2024-05-20 00:19:31 [main] DEBUG i.m.context.DefaultBeanContext  Finding candidate beans for type: Application
console: 2024-05-20 00:19:31 [main] DEBUG i.m.context.DefaultBeanContext  No bean candidates found for type: Application
console: 2024-05-20 00:19:31 [main] DEBUG i.m.context.condition.Condition Bean [Definition: io.micronaut.jaxrs.runtime.core.JaxRsRuntime] will not be loaded due to failing conditions:
```

**This works only if I do a mvn clean install from command prompt and then click run in IntelliJ.** 

Following is my compiler plugin to provide the annotation processors. 

    ```
       <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>${compiler.plugin.version}</version>
                <configuration>
                    <forceJavacCompilerUse>true</forceJavacCompilerUse>
                    <source>${jdk.version}</source>
                    <target>${release.version}</target>
                    <encoding>UTF-8</encoding>
                    <compilerArgs>
                        <arg>-Xlint:all</arg>
                        <arg>-XDcompilePolicy=simple</arg>
                    </compilerArgs>
                    <annotationProcessorPaths combine.children="override">
                        <path>
                            <groupId>org.projectlombok</groupId>
                            <artifactId>lombok</artifactId>
                            <version>${lombok.version}</version>
                        </path>
                        <path>
                            <groupId>io.micronaut.security</groupId>
                            <artifactId>micronaut-security-annotations</artifactId>
                        </path>
                        <path>
                            <groupId>io.micronaut</groupId>
                            <artifactId>micronaut-inject-java</artifactId>
                        </path>
                        <path>
                            <groupId>io.micronaut.jaxrs</groupId>
                            <artifactId>micronaut-jaxrs-processor</artifactId>
                        </path>
                        <path>
                            <groupId>io.micronaut</groupId>
                            <artifactId>micronaut-http-validation</artifactId>
                        </path>
                        <path>
                            <groupId>io.micronaut</groupId>
                            <artifactId>micronaut-graal</artifactId>
                        </path>
                        <path>
                            <groupId>io.micronaut.openapi</groupId>
                            <artifactId>micronaut-openapi</artifactId>
                        </path>
                        <path>
                            <groupId>io.micronaut.validation</groupId>
                            <artifactId>micronaut-validation-processor</artifactId>

                        </path>
                        <path>
                            <groupId>io.micronaut.tracing</groupId>
                            <artifactId>micronaut-tracing-opentelemetry-annotation</artifactId>
                        </path>
                    </annotationProcessorPaths>
                </configuration>
            </plugin>

```

**Comment by graemerocher** [comment]
make sure you enable annotation processors in IntelliJ

**Reply by subash89** [reply]
@graemerocher  Annotation Processor is enabled.

<img width="740" alt="Screenshot 2024-05-20 at 8 27 33 AM" src="https://github.com/micronaut-projects/micronaut-core/assets/703986/77697885-95ca-49fa-9e0e-83c3ca52992d">

**Comment by graemerocher** [comment]
what are the failing conditions. See message `will not be loaded due to failing conditions:`

**Reply by subash89** [reply]
@graemerocher  Following is one additional log I see.

`console: 2024-05-20 00:19:31 [main] DEBUG i.m.context.condition.Condition * No bean of type [jakarta.ws.rs.core.Application] present within context`

Looks like IntelliJ cannot find jakarta.ws.rs.core.Application. This has to be part of the micronaut jaxrs annotation processors that add this parent class to my resource class to make it a jaxrs app.

**Reply by subash89** [reply]
@graemerocher  with annotation processor being enabled, how come only after I do a mvn build, then IntelliJ can pick it up. (I tried with IntelliJ latest Ultimate and CE both) same result.

**Reply by graemerocher** [reply]
if `jakarta.ws.rs.core.Application` is missing then you are classpath is wrong. Check that this class is on your classpath when running from IntelliJ

**Reply by subash89** [reply]
@graemerocher that's already there.
![Screenshot 2024-05-21 at 11 14 02 AM](https://github.com/micronaut-projects/micronaut-core/assets/703986/3b5b2f75-e4df-411d-a975-cd6a023cabf2)

**Reply by subash89** [reply]
But in IntelliJ, target/classes/com/..../.  **I cannot see Micronaut generated classes(starts with $) added in there**.

**Reply by subash89** [reply, marked as answer]
This helped.
[
<img width="661" alt="Screenshot 2024-05-22 at 11 56 04 PM" src="https://github.com/micronaut-projects/micronaut-core/assets/703986/18fe6578-90b4-4875-8809-f986a4bf5bd5">
](url)
